### ISSUE TITLE
How to deal with using requests and urllib3 together

### ISSUE THREAD

**Issue by sjmh** [kennethreitz/requests#2867]
Ran into an interesting issue in the past week while testing things out with SaltStack.  ( SaltStack is a remote execution engine / cfg mgmt tool ).  SaltStack uses requests internally.

However, one of the 'modules' you can use is python-etcd.  python-etcd uses urllib3.  But, because of request's wrapping of urllib3 exceptions, some of the python-etcd code bombs because it tries to catch urllib3 exceptions via fully-qualified names.

An example of this happening can be easily demo'd with:

```
#!/usr/bin/python

#import requests
import etcd

c = etcd.Client('127.0.0.1', 4001)
c.read("/blah", wait=True, timeout=2)
```

Without Requests, you get etcd.EtcdConnectionFailed.  With it, you get urllib3.exceptions.ReadTimeoutError.  python-etcd tries to catch urllib3 exceptions, but because of request's wrapping, it fails to catch them.

Is there a way to work around this properly ( besides going to the python-etcd author and asking him to change everything to use requests instead of urllib3 )?

**Comment by sigmavirus24** [comment]
How are you installing requests and urllib3?

**Comment by sjmh** [comment]
Via RPM ( both from RHEL's official repos for RHEL6 )

```
[root@alpha site-packages]# rpm -qa | grep urllib3
python-urllib3-1.10.2-1.el6.noarch
[root@alpha site-packages]# rpm -qa | grep requests
python-requests-2.6.0-3.el6.noarch
```

**Comment by sigmavirus24** [comment]
So this is a version of requests that had some magic handling for downstream redistributors that remove parts of requests. This should be fixed in newer versions of requests but I have no control over RHEL's repos.

**Comment by sjmh** [comment]
@sigmavirus24 No problem - the pip versions should be good though?

**Comment by sigmavirus24** [comment]
@sjmh should be!

**Comment by sjmh** [comment]
@sigmavirus24 Alrighty - thanks, just confirmed those versions work! Do you know if there's any way to get around the issue with those RPM versions?  If not, is there a minimum version that this issue would be fixed in ( Looks like newest from pip is 2.8.1 ) or is this a general thing for all downstream redistributors in any version?

Thanks for the help.

**Comment by sigmavirus24** [comment]
I forget in which version we fixed it upstream but I'm tempted to say 2.7.0.
